### PR TITLE
Issue2204 gmt mbl cooling components

### PR DESCRIPTION
**Pull request by JingWang-CUB**
@khinkelman could you please perform a minimum review for these cooling models? Please let me know if you have any questions.

### CODE REVIEW

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/CoolingTowerWithBypass.mo [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+    "Cooling tower system"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage valByp(
+    redeclare final package Medium=Medium,
+    final m_flow_nominal=m_flow_nominal*0.0001,
+    final dpValve_nominal=dp_nominal,
+    final use_inputFilter=use_inputFilter)
+    "Condenser water bypass valve"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}},origin={0,-40})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTCWSup(
```
Comment (khinkelman) [inline Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/CoolingTowerWithBypass.mo:129]: Please add description to block

File: Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mos [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	filename="ChilledWaterPumpSpeed.mat", 
+	colors={{28,108,200}}, 
+	displayUnits={"Pa"});
+createPlot(
+	id=1, 
+	position={71, -1, 892, 547}, 
+	y={"chiWatPumSpe.masFloPum"}, 
+	range={0.0, 1200.0, 0.2, 0.6000000000000001}, 
+	grid=true, 
+	subPlot=102,
```
Comment (khinkelman) [inline Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mos:24]: Change subplot to 2

File: Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mos [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	subPlot=102, 
+	colors={{28,108,200}}, 
+	displayUnits={"kg/s"});
+createPlot(
+	id=1, 
+	position={71, -1, 892, 547}, 
+	y={"chiWatPumSpe.y[1]", "chiWatPumSpe.y[2]"}, 
+	range={0.0, 1200.0, -1.0, 2.0}, 
+	grid=true, 
+	subPlot=103,
```
Comment (khinkelman) [inline Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mos:33]: Change subplot to 3

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/package.mo [hunk 4]
```diff
@@ -0,0 +1,5 @@
+within Buildings.Applications.DHC.CentralPlants.Cooling.Controls;
```
Comment (khinkelman) [inline Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/package.mo:1]: Please change `within` path to `Experimental`

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/package.mo [hunk 5]
```diff
@@ -0,0 +1,5 @@
+within Buildings.Applications.DHC.CentralPlants.Cooling.Subsystems;
```
Comment (khinkelman) [inline Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/package.mo:1]: Please change `within` path to `Experimental`

File: Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerParallel.mos [hunk 6]
```diff
@@ -0,0 +1,39 @@
+﻿// Simulate
```
Comment (khinkelman) [inline Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerParallel.mos:1]: The unit tests produced slightly different results than the reference results for this model. Could you please verify the correct results and update the .txt file as needed?
image

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mo [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    "Measured demand side pressure difference"
+    annotation (Placement(transformation(extent={{-60,-40},{-40,-20}})));
+  Modelica.Blocks.Sources.Pulse mFloTot(
+    amplitude=0.4*chiWatPumSpe.m_flow_nominal,
+    period=300,
+    offset=0.5*chiWatPumSpe.m_flow_nominal,
+    startTime=150)
+    "Total chilled water mass flow rate"
+    annotation (Placement(transformation(extent={{-60,20},{-40,40}})));
+equation
```
Comment (khinkelman) [inline Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mo:26]: Please add `initial equation` section with warning message for beta version

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChillerStage.mo [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      each displayUnit="s")={300,900})
+    "On signal of the cooling plant"
+    annotation (Placement(transformation(extent={{-60,20},{-40,40}})));
+  Modelica.Blocks.Sources.Sine QTot(
+    amplitude=0.5*chiStaCon.QEva_nominal,
+    freqHz=1/300,
+    offset=0.5*chiStaCon.QEva_nominal)
+    "Total cooling load"
+    annotation (Placement(transformation(extent={{-60,-40},{-40,-20}})));
+equation
```
Comment (khinkelman) [inline Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChillerStage.mo:21]: Please add `initial equation` section with warning message for beta version

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerParallel.mo [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+    "Weather data bus"
+    annotation (Placement(transformation(extent={{-80,80},{-60,100}})));
+  Buildings.Fluid.Movers.FlowControlled_m_flow pum(
+    redeclare package Medium=Medium,
+    m_flow_nominal=m_flow_nominal,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial,
+    nominalValuesDefineDefaultPressureCurve=true)
+    "Pump for chilled water loop"
+    annotation (Placement(transformation(extent={{-40,40},{-20,60}})));
+equation
```
Comment (khinkelman) [inline Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerParallel.mo:103]: Please add `initial equation` section with warning message for beta version

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerWithBypass.mo [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+    annotation (Placement(transformation(extent={{-40,40},{-20,60}})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTCWLvg(
+    redeclare final package Medium=Medium,
+    m_flow_nominal=m_flow_nominal)
+    "Sensor for leaving conderser water temperature"
+    annotation (Placement(transformation(extent={{30,40},{50,60}})));
+  Buildings.BoundaryConditions.WeatherData.Bus weaBus
+    "Weather data bus"
+    annotation (Placement(transformation(extent={{-80,80},{-60,100}})));
+equation
```
Comment (khinkelman) [inline Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerWithBypass.mo:89]: Please add `initial equation` section with warning message for beta version

**Comment by khinkelman** [comment]
Hi @JingWang-CUB , please address the comments above and update the PR. For reference, these new models were able to simulate in JModelica with warnings but no errors. Below is a summary of the review checklist and current status. The unchecked items are associated with the comments above. Please let me know if you have any questions. 

**Minimum Technical Review - Status Summary:**
- [X] Follow the LBNL Style Guide, User Guide Best Practices, and User Guide Conventions 
- [ ] The models contain required beta warnings
- [ ] Branch passes CI testing 
        @AntoineGautier : It is unclear to me why the CI build is failing. Any suggestions? I ran JModelica without errors, but maybe it is because 1 of the reference results was different with the unit tests? Your input is much appreciated.
- [X] Verify that there are no unintended changes 
- [X] Block diagrams are used to compute variables as much as possible
- [ ] Each instance has a description string
- [X] Each model has a `defaultComponentName`, except the base classes
- [X] A minimum documentation is provided for each class.
- [X] The main models and blocks have a corresponding validation or example model  
- [X] The main models have a simulation script with reference results stored in the library
- [X] Simulations pass in JModelica 
- [X] modelica-fmt was run on the modified packages
- [X] The full path of any instantiated component is used
- [X] Extensible developed on modelica level
- [ ] Run unit tests

**Comment by AntoineGautier** [comment]
It is unclear to me as well. I just pushed a dummy commit to trigger the CI testing. Let's see.

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/CoolingTowerWithBypass.mo (hunk 1 above)
Comment (khinkelman) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/CoolingTowerWithBypass.mo:129]: Done

File: Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mos (hunk 2 above)
Comment (khinkelman) [reply Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mos:24]: Done

File: Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mos (hunk 3 above)
Comment (khinkelman) [reply Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mos:33]: Done

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/package.mo (hunk 4 above)
Comment (khinkelman) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/package.mo:1]: Done

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/package.mo (hunk 5 above)
Comment (khinkelman) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/package.mo:1]: Done

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mo (hunk 7 above)
Comment (khinkelman) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChilledWaterPumpSpeed.mo:26]: Done

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChillerStage.mo (hunk 8 above)
Comment (khinkelman) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Controls/Validation/ChillerStage.mo:21]: Done

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerParallel.mo (hunk 9 above)
Comment (khinkelman) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerParallel.mo:103]: Done

File: Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerWithBypass.mo (hunk 10 above)
Comment (khinkelman) [reply Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerWithBypass.mo:89]: Done

File: Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerParallel.mos (hunk 6 above)
Comment (khinkelman) [reply Buildings/Resources/Scripts/Dymola/Experimental/DHC/CentralPlants/Cooling/Subsystems/Examples/CoolingTowerParallel.mos:1]: Updated

**Comment by khinkelman** [comment]
PR #2261 completes the above requirements once merged.

**Comment by khinkelman** [comment]
@AntoineGautier, this branch has been reviewed and is ready to be merged after CI tests pass. Please let me know if you need anything else. Thanks

**Comment by AntoineGautier** [comment]
@khinkelman I committed a few fixes. 

- https://github.com/lbl-srg/modelica-buildings/pull/2254/commits/6de950921c1e5cc2924505bd6a2230bc82dbafd6 solves this CI error https://travis-ci.com/github/lbl-srg/modelica-buildings/jobs/444086256#L3368
- However, there are discrepancies with the reference results in another model, see https://travis-ci.com/github/lbl-srg/modelica-buildings/jobs/444086257#L443-L450. I did not overwrite the reference results as there might be an issue with the model. Please validate. Then I can commit the new reference results so that you don't have to open a new PR if you want.

**Comment by khinkelman** [comment]
Hi @AntoineGautier, I pulled your recent commits and was unable to reproduce the warnings you received (please see screenshot below). It appears @JingWang-CUB did however did produce the correct reference results originally, and there might be a problem with my configuration. Please go ahead and overwrite the reference results. 
image

For the two requirements, I would assume @JayHuLBL may have added these? (Please correct me if I'm wrong.) Jianjun, please clarify the second requirement if this was something you had added. To me, removing these two requirements seems reasonable.

**Pull request merged**